### PR TITLE
Remove `pub` from `MockConsensusState::root` field

### LINKED ISSUE #1216: `MockConsensusState` should not have commitment root
## Improvement Summary

[`mock::ConsensusState` from `ibc_proto`](https://docs.rs/ibc-proto/latest/ibc_proto/ibc/mock/struct.ConsensusState.html) doesn't have any commitment root.

## Proposal

To avoid any unexpected assumptions, we should also remove commitment_root from [`MockConsensuState` in `ibc-testkit`](https://docs.rs/ibc-testkit/latest/ibc_testkit/testapp/ibc/clients/mock/consensus_state/struct.MockConsensusState.html). The [`ConsensusState::root`](https://docs.rs/ibc-testkit/latest/ibc_testkit/testapp/ibc/clients/mock/consensus_state/struct.MockConsensusState.html#method.root) implementation should return a constant value.

Ref: https://github.com/cosmos/ibc-rs/issues/1052#issuecomment-1906675569

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #1216 

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: ibc-testkit/src/testapp/ibc/clients/mock/consensus_state.rs [hunk 1]
```diff
@@ -9,11 +9,17 @@ use crate::testapp::ibc::clients::mock::header::MockHeader;
 use crate::testapp::ibc::clients::mock::proto::ConsensusState as RawMockConsensusState;
 pub const MOCK_CONSENSUS_STATE_TYPE_URL: &str = "/ibc.mock.ConsensusState";
 
+// Note: This type differs from the `mock::ConsensusState` type exposed by
+// ibc-proto in a few ways:
+// - ibc-proto's `mock::ConsensusState`'s `header` field takes the form of an
+//   `Option<Header>`, while the `header` field here is not optional.
+// - ibc-proto's `mock::ConsensusState` does not have a `root` field for holding
+//   the `CommitmentRoot`; this field exists here because ...
```
Comment (rnbguy) [inline ibc-testkit/src/testapp/ibc/clients/mock/consensus_state.rs]: Can you make it a Rust doc-comment?

Also, no need to comment on the `Option<Header>` as that's an issue from [protobuf](https://github.com/protocolbuffers/protobuf/issues/2497) itself.

If you may, you can comment on why we _had_ to keep a `root` field and not return a constant at `ConsensusState::root` implementation.

Comment (seanchen1991) [reply ibc-testkit/src/testapp/ibc/clients/mock/consensus_state.rs]: > you can comment on why we had to keep a root field

I'm not sure what the reasoning is for this; do you have the context for this?

Comment (rnbguy) [reply ibc-testkit/src/testapp/ibc/clients/mock/consensus_state.rs]: I have to demonstrate to you. Here is an [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e72fd29a1540a6a1c20e6e548d51189f).

**Review by rnbguy — APPROVED** [review]
Look good :+1:

**Pull request merged**